Document search apis and remaining config options (#1003)
grafana/tempo: docs/tempo/website/api_docs/_index.md

By default this endpoint returns [OpenTelemetry](https://github.com/open-telemetry/opentelemetry-proto/tree/main/opentelemetry/proto/trace/v1) JSON,
but if it can also send OpenTelemetry proto if `Accept: application/protobuf` is passed.

> ### Search

Reviewer: I think we should mention here that this is gated behind a feature flag and also we should mark all of this experimental/subject to change. I don't think we want to give 1.x stability guarantees on these endpoints.

Author: Add similar color and phrasing as Loki experimental apis: https://grafana.com/docs/loki/latest/api/#list-rule-groups

Tempo's Search API finds traces based on span and process attributes (tags and values).  The API is available in the query frontend service in
a microservices deployment, or the Tempo endpoint in a single binary deployment.  The following request is used to find traces containing spans
from service "myservice" and the url contains "api/myapi".

```
GET /api/search?service.name=myservice&http.url=api/myapi
```

Each query parameter is of the form <name>=<value>, where <name> is the name of any span-level or process-level attribute.  The value is matched as a case-insenstive substring.  There are several reserved query parameters:
- `minDuration=<go duration value>`
  Optional.  Find traces with at least this duration.  Duration values are of the form `10s` for 10 seconds, `100ms`, `30m`, etc.
- `maxDuration=<go duration value>`

>   Optional.  Find traces with no greater than this duration.  Same format as `minDuration`

Reviewer 2: `limit` is also a reserved query parameter

By default this endpoint returns [OpenTelemetry](https://github.com/open-telemetry/opentelemetry-proto/tree/main/opentelemetry/proto/trace/v1) JSON,
but if it can also send OpenTelemetry proto if `Accept: application/protobuf` is passed.

### Search

<span style="background-color:#f3f973;">This experimental endpoint is disabled by default and can be enabled via the search_enabled YAML config option.</span>

Tempo's Search API finds traces based on span and process attributes (tags and values).  The API is available in the query frontend service in
a microservices deployment, or the Tempo endpoint in a single binary deployment.  The following request is used to find traces containing spans
from service "myservice" and the url contains "api/myapi".

```
GET /api/search?service.name=myservice&http.url=api/myapi
```

> Each query parameter is of the form <name>=<value>, where <name> is the name of any span-level or process-level attribute.  The value is matched as a case-insenstive substring.  There are several reserved query parameters:

Reviewer 3: Suggestion corrects spelling.

I have a further question:  what is a "reserved" query parameter? Would it clarify to say that these are defined parameters?
 
```suggestion
Each query parameter is of the form <name>=<value>, where <name> is the name of any span-level or process-level attribute.  The value is matched as a case-insensitive substring.  There are several reserved query parameters:
```

Author: This API was created generically, and uses query parameters of the form `name=value (&name2=value2...)` In this case there is no distinction between the generic parameters and reserved names, i.e. `limit` is interpreted as a special value, and not as a value to search for.

- `minDuration = (go duration value)`
  Optional.  Find traces with at least this duration.  Duration values are of the form `10s` for 10 seconds, `100ms`, `30m`, etc.
- `maxDuration = (go duration value)`

>   Optional.  Find traces with no greater than this duration.  Same format as `minDuration`

Reviewer 3: ```suggestion
  Optional.  Find traces with no greater than this duration.  Uses the same form as `minDuration`.
```

- `limit = (integer)`
  Optional.  Limit the number of search results. Default is 100.

### Search Tags

<span style="background-color:#f3f973;">This experimental endpoint is disabled by default and can be enabled via the search_enabled YAML config option.</span>

> This api retrieves all discovered tag names that can be used in search.  The API is available in the query frontend service in

Reviewer 3: ```suggestion
This endpoint retrieves all discovered tag names that can be used in search.  The endpoint is available in the query frontend service in
```

a microservices deployment, or the Tempo endpoint in a single binary deployment.

```
GET /api/search/tags
```

### Search Tag Values

<span style="background-color:#f3f973;">This experimental endpoint is disabled by default and can be enabled via the search_enabled YAML config option.</span>

> This API retrieves all discovered values for the given tag, that can be used in search.  The API is available in the query frontend service in

Reviewer 3: ```suggestion
This endpoint retrieves all discovered values for the given tag, which can be used in search.  The endpoint is available in the query frontend service in
```
